tpm2/tpm2_test.go

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tpm2
+
+import (
+	"bytes"
+	"flag"
+	"os"
+	"testing"
+)
+
+var runIntegration = flag.Bool("integration", false, "Run integration tests using /dev/tpm0")
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	os.Exit(m.Run())
+}
+
+func TestGetRandom(t *testing.T) {
+	if !*runIntegration {
+		t.SkipNow()
+	}
+	rw, err := OpenTPM("/dev/tpm0")
+	if err != nil {
+		t.Fatalf("OpenTPM failed: %s", err)
+	}
+	defer rw.Close()
+
+	_, err = GetRandom(rw, 16)
+	if err != nil {
+		t.Fatalf("GetRandom failed: %v", err)
+	}
+}
+
+func TestReadPCRs(t *testing.T) {
+	if !*runIntegration {
+		t.SkipNow()
+	}
+	rw, err := OpenTPM("/dev/tpm0")
+	if err != nil {
+		t.Fatalf("OpenTPM failed: %s", err)
+	}
+	defer rw.Close()
+
+	pcr := []byte{0x03, 0x80, 0x00, 0x00}
+	_, _, _, _, err = ReadPCRs(rw, pcr)
```

Comment:
For Go style: collapse this statement into the next if statement. Same comment for other tests in this file.

Reply:
Done